File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/PrivateMsgAdapter.java
package com.scatl.uestcbbs.module.message.adapter;

import android.text.TextUtils;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.PrivateMsgBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;
import com.scatl.uestcbbs.util.TimeUtil;

/**
 * author: sca_tl
 * description:
 * date: 2020/1/29 16:25
 */
public class PrivateMsgAdapter extends BaseQuickAdapter<PrivateMsgBean.BodyBean.ListBean, BaseViewHolder> {

    public PrivateMsgAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, PrivateMsgBean.BodyBean.ListBean item) {
        helper.setText(R.id.user_name, item.toUserName)
                .setText(R.id.content, TextUtils.isEmpty(item.lastSummary) ? "[图片]" : item.lastSummary)
                .setText(R.id.time, TimeUtil.formatTime(item.lastDateline, R.string.post_time1, mContext))
                .addOnClickListener(R.id.user_icon);
        GlideLoader4Common.simpleLoad(mContext, item.toUserAvatar, helper.getView(R.id.user_icon));

        helper.getView(R.id.unread_img).setVisibility(item.isNew == 1 ? View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/PrivateMsgAdapter.kt
package com.scatl.uestcbbs.module.message.adapter

import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.databinding.ItemPrivateMsgBinding
import com.scatl.uestcbbs.entity.PrivateMsgBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.TimeUtil
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 14:40
 */
class PrivateMsgAdapter: PreloadAdapter<PrivateMsgBean.BodyBean.ListBean, ItemPrivateMsgBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemPrivateMsgBinding {
        return ItemPrivateMsgBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemPrivateMsgBinding>, position: Int, item: PrivateMsgBean.BodyBean.ListBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.toUserName
        holder.binding.content.text = if (TextUtils.isEmpty(item.lastSummary)) "[图片]" else item.lastSummary
        holder.binding.time.text = TimeUtil.formatTime(item.lastDateline, R.string.post_time1, context)
        holder.binding.userIcon.load(item.toUserAvatar)
        holder.binding.unreadImg.visibility = if (item.isNew == 1) View.VISIBLE else View.GONE
    }
}